src/dmd/dmodule.d

```diff
@@ -1058,15 +1059,26 @@ extern (C++) final class Module : Package
              *
              * To avoid the conflict:
              * 1. If preceding package name insertion had occurred by Package::resolve,
-             *    later package.d loading will change Package::isPkgMod to PKG.module_ and set Package::mod.
+             *    reuse the previous wrapping 'Package' if it exists
              * 2. Otherwise, 'package.d' wrapped by 'Package' is inserted to the internal tree in here.
+             *
+             * Then change Package::isPkgMod to PKG.module_ and set Package::mod.
              */
-            auto p = new Package(Loc.initial, ident);
+            auto p = cast(Package)dst.lookup(ident);
+            if (p)
+            {
+                this.tag = p.tag; // reuse the 'older' package tag
+            }
+            else
+            {
+                p = new Package(Loc.initial, ident);
+                p.tag = this.tag; // reuse the same package tag
+                p.symtab = new DsymbolTable();
+            }
```

Comment:
Only the else case actually needs to be there. It'll simplify the logic.

Reply:
Done.

Reply:
much better